sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SharedThriftServer.scala

```diff
@@ -138,10 +138,15 @@ trait SharedThriftServer extends SharedSparkSession {
     sqlContext.setConf(ConfVars.HIVE_START_CLEANUP_SCRATCHDIR.varname, "true")
 
     try {
-      hiveServer2 = HiveThriftServer2.startWithContext(sqlContext)
+      hiveServer2 = HiveThriftServer2.startWithContext(sqlContext, exitOnError = false)
       hiveServer2.getServices.asScala.foreach {
         case t: ThriftCLIService =>
           serverPort = t.getPortNumber
+          if (t.isInstanceOf[ThriftBinaryCLIService] && mode == ServerMode.http) {
+            logError("A previous Hive's SessionState is leaked, aborting this retry")
+            throw new IllegalStateException("HiveThriftServer2 started in binary mode " +
```

Comment:
shall we use `SparkException.internalError`?

Reply:
addressed